providers/os/connection/snapshot/blockdevices.go

```diff
@@ -126,7 +126,9 @@ func (blockEntries BlockDevices) GetMountablePartitionByDevice(device string) (*
 	return &PartitionInfo{Name: devFsName, FsType: partitions[0].FsType}, nil
 }
 
-func lms(lmsCache map[string]int, s1, s2 string) int {
+// LongestMatchingSuffix returns the length of the longest common suffix of two strings
+// and caches the result (lengths of the matching suffix) for future calls with the same string
+func LongestMatchingSuffix(lmsCache map[string]int, s1, s2 string) int {
```

Comment:
curious: do we need the caching? whats the use case for trying to fetch this twice for the same string?

Reply:
We compare the lengths of two findings in the loop (because FindDevice actually sorts the devices per lms), and then we check it one more time in the end
LongestMatchingSuffix does iterate over devices names itself
I get it that this function isn't called often and strings aren't that long, but just why make it slower from start